sympy/calculus/codomain.py

```diff
@@ -151,3 +154,83 @@ def codomain_interval(f, set_val, *sym):
     if isinstance(domain, FiniteSet):
         return FiniteSet(*[limit(func, symbol, i) if i in FiniteSet(-oo, oo)
                             else func.subs({symbol: i}) for i in domain])
+
+
+def not_empty_in(finite_set, sets, *syms):
```

Comment:
This is not quite in the way we want it. Say you have an unevaluated intersection produced by `solveset`, for example `FiniteSet(x).intersect(Interval(0, 1))` for such sets the fact that the given set is empty or non-empty depends on the value of `x` and with your `not_empty_in` method we want to tell for what values of `x` this set will be non-empty. So, the input should be an unevaluated intersection and a symbol(s) and it should return another set.